core/templates/dev/head/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -17,72 +17,91 @@
  * contributors to contribute.
  */
 
-require('domain/opportunity/ExplorationOpportunitySummaryObjectFactory.ts');
-require('domain/opportunity/SkillOpportunityObjectFactory.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import {
+  UrlInterpolationService
+} from 'domain/utilities/url-interpolation.service';
+import {
+  ExplorationOpportunitySummary
+} from 'domain/opportunity/ExplorationOpportunitySummaryObjectFactory';
+import {
+  SkillOpportunity
+} from 'domain/opportunity/SkillOpportunityObjectFactory';
 
-angular.module('oppia').factory('ContributionOpportunitiesBackendApiService', [
-  '$http', 'ExplorationOpportunitySummaryObjectFactory',
-  'SkillOpportunityObjectFactory', 'UrlInterpolationService',
-  'OPPORTUNITY_TYPE_SKILL', 'OPPORTUNITY_TYPE_TRANSLATION',
-  'OPPORTUNITY_TYPE_VOICEOVER',
-  function($http, ExplorationOpportunitySummaryObjectFactory,
-      SkillOpportunityObjectFactory, UrlInterpolationService,
-      OPPORTUNITY_TYPE_SKILL, OPPORTUNITY_TYPE_TRANSLATION,
-      OPPORTUNITY_TYPE_VOICEOVER) {
-    var urlTemplate = '/opportunitiessummaryhandler/<opportunityType>';
-    var _getOpportunityFromDict = function(opportunityType, opportunityDict) {
-      if (
-        opportunityType === OPPORTUNITY_TYPE_VOICEOVER ||
-        opportunityType === OPPORTUNITY_TYPE_TRANSLATION) {
-        return ExplorationOpportunitySummaryObjectFactory.createFromBackendDict(
-          opportunityDict);
-      } else if (opportunityType === OPPORTUNITY_TYPE_SKILL) {
-        return SkillOpportunityObjectFactory.createFromBackendDict(
-          opportunityDict);
+@Injectable({
+  providedIn: 'root'
+})
+export class ContributionOpportunitiesBackendApiService {
+  urlTemplate = '/opportunitiessummaryhandler/<opportunityType>';
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private http: HttpClient
+  ) {}
+  /**  TODO(srijanreddy98): for some reason not able to import types (
+  * OPPORTUNITY_TYPE_SKILL,
+  * OPPORTUNITY_TYPE_TRANSLATION,
+  * OPPORTUNITY_TYPE_VOICEOVER
+  * )from constants.ts in assets */
+  _getOpportunityFromDict(opportunityType, opportunityDict) {
```

Comment:
should we add `type` for function and params?

Reply:
Ditto elsewhere*

Reply:
Done